src/pspc/sweep/BasisFieldState.tpp

```diff
@@ -116,7 +116,7 @@ namespace Pspc
    * Set System state to current state of the BasisFieldState object.
    */
    template <int D>
-   void BasisFieldState<D>::setSystemState(bool isFlexible)
+   void BasisFieldState<D>::setSystemState(bool newCellParams)
```

Comment:
Please explain the motivation and meaning of this change in name to me when you get a chance.

Reply:
The reason for the new name is that there are now 2 ways that the unit cell parameters can be changed after a Sweep step:

1. The lattice parameters are flexible, so they are updated each iteration
2. The user is performing a sweep that changes the value of one or more lattice parameters, so the lattice parameters are different on each step along the sweep.

If either of these are true (or both are true), we need to update the unit cell. So, rather than using isFlexible as the criterion for resetting the unit cell, I added a new loop in Sweep.tpp that checks whether any of the lattice parameters have changed since the last step of the sweep, hence "newCellParams". This should be set to true if either `isFlexible() == true` or if we are sweeping a lattice parameter.